src/autopas/containers/octree/OctreeNodeWrapper.h

```diff
@@ -0,0 +1,193 @@
+/**
+ * @file OctreeNodeWrapper.h
+ *
+ * @author Johannes Spies
+ * @date 03.05.2021
+ */
+#pragma once
+
+#include <memory>
+
+#include "autopas/containers/octree/OctreeNodeInterface.h"
+#include "autopas/iterators/ParticleIteratorWrapper.h"
+#include "autopas/utils/ArrayMath.h"
+#include "autopas/utils/WrapOpenMP.h"
+
+namespace autopas {
+/**
+ * This class wraps the functionality provided by the octree leaves and inner nodes in a structure that adheres to the
+ * ParticleCell concept. This is necessary to use the octree nodes as particle containers.
+ *
+ * @tparam Particle The particle class that should be used in the octree cell.
+ */
+template <typename Particle>
+class OctreeNodeWrapper : public ParticleCell<Particle> {
```

Comment:
In my opinion this would not necessarily lead to cleaner code: If I inherited from `OctreeNodeInterface<Particle>` in `OctreeNodeWrapper`, I would have to implement the virtual methods that are defined in the `OctreeNodeInterface<Particle>` as well (for instance `insert`). However, this interface should not be exposed to the outside. (`insert` returns a pointer to the child itself, which should only be used internally by the octree container. In my opinion, this is solved by using the wrapper.)

I think you are very right that a method calling the same method does not make some beautiful code. But in my opinion, compared to the downside of having to implement the virtual method from `OctreeNodeInterface<Particle>` in the `OctreeNodeWrapper` (and let those only throw an exception whenever called), this is the smaller evil.

Reply:
Ok sounds reasonable. However, I think it would also be reasonable to explain this reasoning in the documentation of the class ;)

Reply:
I added the reasoning to the documentation in d7a2c2c:

https://github.com/AutoPas/AutoPas/blob/d7a2c2c349a1eb571268ebf72b92d489f316cbb5/src/autopas/containers/octree/OctreeNodeWrapper.h#L17-L37

Let me know if this works for you! :)